test/CMakeLists.txt

```diff
@@ -984,179 +1021,238 @@ if(iodaconv_bufr_ENABLED)
-                    ARGS    ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
-                            netcdf
-                    "${CMAKE_BINARY_DIR}/bin/bufr2ioda.x testinput/satwind_EUMet_wmoBUFR2ioda.yaml"
-                    satwind_EUMet.nc ${IODA_CONV_COMP_TOL_ZERO}
+                    COMMAND ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
+                    ARGS    netcdf
+                    "${CMAKE_BINARY_DIR}/bin/bufr2ioda.x testinput/bufr_empty_fields.yaml"
+                    bufr_empty_fields.nc ${IODA_CONV_COMP_TOL_ZERO}
                     DEPENDS bufr2ioda.x )
 
-  ecbuild_add_test( TARGET  test_iodaconv_bufr_satwind_Himawari_wmo_bufr
+  ecbuild_add_test( TARGET  test_iodaconv_bufr_sfcshp
                     TYPE    SCRIPT
-                    COMMAND bash
-                    ARGS    ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
-                            netcdf
-                    "${CMAKE_BINARY_DIR}/bin/bufr2ioda.x testinput/satwind_Himawari_wmoBUFR2ioda.yaml"
-                    satwind_Himawari.nc ${IODA_CONV_COMP_TOL_ZERO}
+                    COMMAND ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
+                    ARGS    netcdf
+                    "${CMAKE_BINARY_DIR}/bin/bufr2ioda.x testinput/bufr_sfcshp.yaml"
+                    bufr_sfcshp.nc ${IODA_CONV_COMP_TOL_ZERO}
                     DEPENDS bufr2ioda.x )
 
-  ecbuild_add_test( TARGET  test_iodaconv_bufr_satwind_Insat_wmo_bufr
+  ecbuild_add_test( TARGET  test_iodaconv_bufr_adpupa
                     TYPE    SCRIPT
                     COMMAND bash
-                    ARGS    ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
-                            netcdf
-                    "${CMAKE_BINARY_DIR}/bin/bufr2ioda.x testinput/satwind_Insat_wmoBUFR2ioda.yaml"
-                    satwind_Insat.nc ${IODA_CONV_COMP_TOL_ZERO}
+                    COMMAND ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
+                    ARGS    netcdf
+                    "${CMAKE_BINARY_DIR}/bin/bufr2ioda.x testinput/adpupa_prepbufr.yaml"
+                    adpupa_prepbufr.nc ${IODA_CONV_COMP_TOL_ZERO}
                     DEPENDS bufr2ioda.x )
 
-  ecbuild_add_test( TARGET  test_iodaconv_bufr_vadwinds_wmo_bufr
+ ecbuild_add_test( TARGET  test_iodaconv_bufr_ncep_sevcsr
                     TYPE    SCRIPT
-                    COMMAND bash
-                    ARGS    ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
-                            netcdf
-                    "${CMAKE_BINARY_DIR}/bin/bufr2ioda.x testinput/vadwinds_wmoBUFR2ioda.yaml"
-                    vadwinds_wmo_multi.nc ${IODA_CONV_COMP_TOL_ZERO}
+                    COMMAND ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
+                    ARGS    netcdf
+                    "${CMAKE_BINARY_DIR}/bin/bufr2ioda.x testinput/bufr_ncep_sevcsr.yaml"
+                    gdas.t00z.sevcsr.tm00.nc ${IODA_CONV_COMP_TOL_ZERO}
                     DEPENDS bufr2ioda.x )
+
+#  ecbuild_add_test( TARGET  test_iodaconv_bufr_aircar
```

Comment:
Why are all these commented out?

Reply:
Yeah I'm with you, 'unit tests' should only be testing code, for science eval, these things should be done offline. JEDI as a whole has way too many unit tests already.

Reply:
I second the concern about test organization. Unit testing is for code coverage (not science verification) and should consist of a set of tiny, fast executing tests (since we run unit tests after every commit push). 

A set of tests for science verification should be outside the set for unit testing. There can be many small or large tests, but we would then run these nightly or on the weekend so that we can accommodate longer runtimes. I think roughly this is what test tier2 is for. @mer-a-o would you agree?

We also need a "system integration" set of tests. This would be more in the vein of running experiments in EWOK and other workflow systems that our partners are using. Ie, validating that all the pieces play together as intended. This can cover science verification, but it also serves the purpose of making sure the interfaces are working properly and for flushing out (and fixing) memory and runtime performance issues. I think this is roughly what test tier3 is for.